packages/sdk/src/evm/core/classes/contract-publisher.ts

```diff
@@ -541,4 +545,19 @@ export class ContractPublisher extends RPCConnectionHandler {
       metadataUri: contractModel.publishMetadataUri,
     });
   }
+
+  private cleanupOldPublishFlowData(
+    extraMetadata: ExtraPublishMetadata,
+  ): ExtraPublishMetadata {
+    return {
+      ...extraMetadata,
+      isDeployableViaFactory: false,
+      isDeployableViaProxy: false,
+      factoryDeploymentData: {
+        ...extraMetadata.factoryDeploymentData,
+        implementationAddresses: {},
+        factoryAddresses: {},
```

Comment:
we def don't use either of these?

Reply:
We use these when deploying, if these exist.

But the new publish relies on `deployType: autoFactory` etc, and doesn't have implementation/factory addresses because of deterministic deployments.

So, if an old published contract has not been republished, it will still get deployed. But, once republished, all the old fields will be reset in favour of new publish flow.